File: Isomorphic Strings.js
function isIsomorphic(s, t) {
    const hashM = {};
    const hashM2 = {};

    for (let i = 0; i < s.length; i++) {
        const a = s[i];
        const b = t[i];

        // Check if the mapping exists and is consistent
        if ((hashM[a] !== undefined && hashM[a] !== b) || (hashM2[b] !== undefined && hashM2[b] !== a)) {
            return false;
        }

        // Update the mappings
        hashM[a] = b;
        hashM2[b] = a;
    }

    return true;
}

File: Isomorphic Strings.py
def isIsomorphic(s, t):
    hash_m = {}
    hash_m2 = {}

    for a, b in zip(s, t):
        # Check if mapping exists and is consistent
        if (a in hash_m and hash_m[a] != b) or (b in hash_m2 and hash_m2[b] != a):
            return False
        hash_m[a] = b
        hash_m2[b] = a

    return True